lib/matplotlib/collections.py

```diff
@@ -1242,15 +1242,16 @@ def set_verts(self, verts, closed=True):
             return
 
         # Fast path for arrays
-        if isinstance(verts, np.ndarray) and len(verts.shape) == 3:
+        if isinstance(verts, np.ndarray) and len(verts.shape) == 3 and verts.size:
             verts_pad = np.concatenate((verts, verts[:, :1]), axis=1)
             # Creating the codes once is much faster than having Path do it
             # separately each time by passing closed=True.
```

Comment:
I believe the comment is not quite good anymore. It was before when the `codes` were explicitly build. No it should be more along the lines "we make one template path and reuse parts of that"

Reply:
makes sense, updated